### PR TITLE
[SMALLFIX]Refactor SessionCleaner class

### PR DESCRIPTION
**Pull request by maobaolong**
This change avoid use sub class to extends the cleanupSession function.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/ISessionCleanup.java [hunk 1]
```diff
@@ -12,11 +12,13 @@
 package alluxio.worker;
 
 /**
- * Interface for classes which track sessions and provide a method to clean any expired sessions.
+ * Interface representing cleanup a special session.
  */
-public abstract class SessionCleanupCallback {
+public interface ISessionCleanup {
```
Comment (calvinjia) [inline core/server/worker/src/main/java/alluxio/worker/ISessionCleanup.java]: We do not use the `I` prefix convention for interfaces.

Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/ISessionCleanup.java]: Done.

**Comment by maobaolong** [comment]
@calvinjia Thank you point this code style. PTAL.

File: core/server/worker/src/main/java/alluxio/worker/SessionCleanup.java [hunk 2]
```diff
@@ -12,11 +12,13 @@
 package alluxio.worker;
 
 /**
- * Interface for classes which track sessions and provide a method to clean any expired sessions.
+ * Interface representing cleanup a special session.
  */
-public abstract class SessionCleanupCallback {
+public interface SessionCleanup {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleanup.java]: NIT: I would call this interface `SessionCleanable`

File: core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java [hunk 3]
```diff
@@ -30,20 +31,23 @@
...
+  private final SessionCleanup mSessionCleanup;
+
   /** Flag to indicate if the checking should continue. */
   private volatile boolean mRunning;
 
   /**
    * Creates a new instance of {@link SessionCleaner}.
-   *
-   * @param sessionCleanupCallback the session clean up callback which will periodically be invoked
+   * @param sessions the session clean up callback which will periodically be invoked
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java]: is the javadoc for `sessions` here still accurate?

File: core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java [hunk 4]
```diff
@@ -30,20 +31,23 @@
...
+   * @param sessionCleanup who want to cleanup the session
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java]: want -> wants

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java [hunk 5]
```diff
@@ -198,19 +197,7 @@ public void start() throws IOException {
...
-      @Override
-      public void cleanupSessions() {
-        for (long session : mSessions.getTimedOutSessions()) {
-          mSessions.removeSession(session);
-          mBlockStore.cleanupSession(session);
-          mUnderFileSystemBlockStore.cleanupSession(session);
-        }
-      }
-    });
+    mSessionCleaner = new SessionCleaner(mSessions, mBlockStore);
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java]: is this correct? 
original code will cleanup session for `mUnderFileSystemBlockStore`, while the code after this PR seems only cleaning up sessions for `mBlockStore`.

File: core/server/worker/src/main/java/alluxio/worker/SessionCleanup.java (hunk 2 above)
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleanup.java]: Done. I think the `SessionCleanable` can make more sense.

File: core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java (hunk 3 above)
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java]: Updated.

File: core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java (hunk 4 above)
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleaner.java]: Thanks for your carefulness and sorry for my carelessness.

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java (hunk 5 above)
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java]: NIT: It is my fault. Now i use varargs to accept more callback.

**Comment by maobaolong** [comment]
@apc999  PTAL.

**Comment by maobaolong** [comment]
@calvinjia Thank you for your review. PTAL.

File: core/server/worker/src/main/java/alluxio/worker/SessionCleanable.java [hunk 6]
```diff
@@ -12,11 +12,13 @@
 package alluxio.worker;
 
 /**
- * Interface for classes which track sessions and provide a method to clean any expired sessions.
+ * Interface representing cleanup a special session.
```
Comment (calvinjia) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleanable.java]: I think the original javadoc still makes sense, what additional information did you want to add?

Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/SessionCleanable.java]: Okey, reverted it.

**Comment by maobaolong** [comment]
@calvinjia PTAL.

**Comment by calvinjia** [comment]
@maobaolong Thanks for improving this part of the code, it is much cleaner now.

**Pull request merged**